experimental/PropertyDDS/packages/property-binder/src/data_binder/dataBinding.ts

```diff
@@ -0,0 +1,1838 @@
+            }
+        } else {
+            // Otherwise the removal of a reference
+            // @ts-ignore
+            let referenceInformation = getInNestedObjects.apply(undefined, [in_referencePropertySubTable].concat(escapeTokenizedPathForMap(in_tokenizedPath)));
+            if (referenceInformation) {
+                this._handleRemovals(
+                    in_tokenizedFullPath,
+                    in_nestedRegisteredPath,
+                    referenceInformation,
+                    level, {
+                    simulated: false,
+                    calledForReferenceTargetChanged: in_calledForReferenceTargetChanged,
+                    removeRootCallbacks: true,
+                    callRootRemovals: false,
+                    callRemovals: true
+                }
+                );
+            }
+        }
+    }
+
+    /**
+     * This function will call the insert, modify and collection* callbacks registered via DataBinding.registerOnPath()
+     * with the appropriate arguments. Additionally, it will take care of binding callbacks to reference properties
+     * to keep track of changes to referenced properties (if a path handler has been registered that traverses the
+     * reference).
+     *
+     * @param in_modificationContext -
+     *     The modifications
+     * @param in_registeredSubPaths -
+     *     The paths for which the user has registered handlers (this structure has to be rooted at the
+     *     same property as the modification context)
+     * @param in_referencePropertySubTable -
+     *     The subtree of the reference property table that is rooted at the same property as the modification context
+     * @param in_calledForReferenceTargetChanged -
+     *     This handler was called due to the target of a reference changing
+     * @param in_rootPathOrProperty -
+     *     Either the property at which the changeSet processed by this function is rooted, or alternatively
+     *     a path to this property. This will be used to resolve other paths.
+     * @param in_referencedPropertyTypeidHolder -
+     *     Object containing full typeid of the referenced object (including context).
+     * @param in_indirectionsAtRoot - The number of indirections at the root of in_referencePropertySubTable
+     * @param in_previousSourceReferencePropertyInfo -
+     *     This contains the reference property table entry for the referencing property. It's used to validate whether
+     *     the currently followed reference chain is still valid.
+     * @param in_tokenizedFullPath -
+     *     The full path from the Data Binding to the current position (including resolved previous references)
+     * @hidden
+     */
+    private _handleModifications(
+        in_modificationContext: ModificationContext,
+        in_registeredSubPaths: any,
+        in_referencePropertySubTable: any,
+        in_calledForReferenceTargetChanged: boolean,
+        in_rootPathOrProperty: any,
```

Comment:
You have similar parameters in  `_handleReferenceModifications` and `_referenceTargetChanged` but types are different